Xamarin.Essentials/Types/Contact.shared.cs

```diff
@@ -1,55 +1,95 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Linq;
 
 namespace Xamarin.Essentials
 {
     public class Contact
     {
-        public string Name { get; }
+        string displayName;
 
-        public ContactType ContactType { get; }
+        public Contact()
+        {
+        }
 
-        public IReadOnlyList<ContactPhone> Numbers { get; }
+        public Contact(
+            string id,
+            string namePrefix,
+            string givenName,
+            string middleName,
+            string familyName,
+            string nameSuffix,
+            IEnumerable<ContactPhone> phones,
+            IEnumerable<ContactEmail> email,
+            string displayName = null)
+        {
+            Id = id;
+            NamePrefix = namePrefix;
+            GivenName = givenName;
+            MiddleName = middleName;
+            FamilyName = familyName;
+            NameSuffix = nameSuffix;
+            Phones.AddRange(phones?.ToList());
+            Emails.AddRange(email?.ToList());
+            DisplayName = displayName;
+        }
 
-        public IReadOnlyList<ContactEmail> Emails { get; }
+        public string Id { get; private set; }
 
-        internal Contact(
-            string name,
-            List<ContactPhone> numbers,
-            List<ContactEmail> email,
-            ContactType contactType)
+        public string DisplayName
         {
-            Name = name;
-            Emails = email;
-            Numbers = numbers;
-            ContactType = contactType;
+            get => !string.IsNullOrWhiteSpace(displayName) ? displayName : $"{GivenName}{GetName(FamilyName)}";
+            private set => displayName = value;
         }
+
+        public string NamePrefix { get; set; }
+
+        public string GivenName { get; set; }
+
+        public string MiddleName { get; set; }
+
+        public string FamilyName { get; set; }
+
+        public string NameSuffix { get; set; }
+
+        public List<ContactPhone> Phones { get; set; } = new List<ContactPhone>();
+
+        public List<ContactEmail> Emails { get; set; } = new List<ContactEmail>();
+
+        string GetName(string name)
+            => string.IsNullOrWhiteSpace(name) ? string.Empty : $" {name}";
     }
 
     public class ContactEmail
     {
-        public string EmailAddress { get; }
-
-        public ContactType ContactType { get; }
+        public ContactEmail()
+        {
+        }
 
-        internal ContactEmail(string email, ContactType contactType)
+        public ContactEmail(string value, ContactEmailType type)
         {
-            EmailAddress = email;
-            ContactType = contactType;
+            Value = value;
+            Type = type;
         }
+
+        public string Value { get; set; }
+
+        public ContactEmailType Type { get; set; }
```

Comment:
As noted in a comment above, removing this for v1.6 is probably best and we can add it back via demand or when doing writing.